### PR TITLE
Closes #7558:  Added pygments entry points for ipython lexers.

### LINKED ISSUE #7558: IPython should specify pygments entry points for its lexers
IPython implements a few pygments lexers - IPyLexer, IPythonTracebackLexer, etc., but does not install pygments entry points, so these lexers cannot be used with the pygmentize command.

Having them available to pygmentize will be helpful because pygmentize has some features that are not available from from nbconvert.

### PR DESCRIPTION
**Pull request by abalkin**
Closes #7558

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -286,7 +286,14 @@ def run(self):
     setup_args['cmdclass']['bdist_wheel'] = css_js_prerelease(get_bdist_wheel())
     
     setuptools_extra_args['zip_safe'] = False
-    setuptools_extra_args['entry_points'] = {'console_scripts':find_entry_points()}
+    setuptools_extra_args['entry_points'] = {
+        'console_scripts': find_entry_points(),
+        'pygments.lexers': [
+            'ipythonconsole = IPython.nbconvert.utils.lexers:IPythonConsoleLexer',
+            'ipython = IPython.nbconvert.utils.lexers:IPythonLexer',
```
Comment (takluyver) [inline setup.py]: Can you add the `ipython2` alias for this as well?

**Comment by takluyver** [comment]
Other than that, :+1: . Entry points are a rare nice bit of setuptools, so we may as well play nicely with them.

**Comment by abalkin** [comment]
@takluyver - I added ipython2.  With this PR, pygmentize reports

```
$ pygmentize -L lexers | grep -i ipython
* ipython2, ipython:
    IPython
* ipython2, ipython:
    IPython
* ipython3:
    IPython3
* ipythonconsole:
    IPython console session
```

Note that I also specified empty filename pattern lists for ipython lexers so that they don't compete with python for '*.py' and friends.

**Comment by abalkin** [comment]
It turned out, aliases don't require a separate entry point.  I reverted that change and now

```
$ pygmentize -L lexers | grep -i ipython
* ipython2, ipython:
    IPython
* ipython3:
    IPython3
* ipythonconsole:
    IPython console session
```

**Comment by takluyver** [comment]
Thanks!

**Pull request merged**